pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERRORS = (select.error, OSError, IOError)
 
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+_SELECT_ERROR_CHECKERS = {
+    select.error : lambda e: e.args[0] == errno.EINTR,
+    OSError      : lambda e: e.errno == errno.EINTR,
+    IOError      : lambda e: e.errno == errno.EINTR,
+    }
```

Comment:
@shinji-s: sorry for the confusion. What I was concerned about is what would happen on python 3.x, where all those different error classes are reduced to the single OSError class. I don't want `select.error : lambda e: e.args[0] == errno.EINTR` to win and be the only one left at the end, because we should be looking at `e.errno` in OSError, not `e.args[0]`. I am hoping that python evaluates the dict properties in order so that in our case, the last one in `_SELECT_ERROR_CHECKERS` would win out.

The experiment in my comment was performed via https://www.python.org/shell/ (python 3.4.0) to demonstrate the collapsing of classes, and which property wins out.

If python guarantees the processing of the dict properties in the specified order, then we should probably have `OSError      : lambda e: e.errno == errno.EINTR` at the end to make that explicit and also add a comment in `_SELECT_ERROR_CHECKERS` that explains that "all these error classes are reduced to the single OSError class in 3.x"